### PR TITLE
Use the query provided by the view for the archive portlet.

### PR DESCRIPTION
**Pull request by maethu**
Based on #53

### CODE REVIEW

**Comment by maethu** [comment]
cc @mbaechtold you may also take a look.

File: ftw/news/portlets/news_archive_portlet.py [hunk 1]
```diff
@@ -75,29 +75,29 @@ def _set_selected_archive(self):
...
-            date)
+        url = '{url}/{view}?archive={date}'.format(
+            url=self.context.absolute_url(),
+            view=self.view.__name__,
+            date=date)
+
+        portlet = self.request.get('portlet', None)
+        manager = self.request.get('manager', None)
+        if portlet:
+            url += '&portlet={0}'.format(portlet)
```
Comment (jone) [inline ftw/news/portlets/news_archive_portlet.py]: Where does the URL go? Should we escape here? -> `&amp;`

File: ftw/news/browser/news_listing.py [hunk 2]
```diff
@@ -174,17 +178,20 @@ def get_manager_and_assignments(self, manager_name):
...
+        if self.portlet:
+            portlet_query = self.portlet.get_query(all_news=True)
+            view_query = super(NewsListingPortlet, self).get_query()
+
+            if 'start' in view_query:
+                portlet_query['start'] = view_query['start']
+            return portlet_query
 
-        return []
+        return {}
```
Comment (jone) [inline ftw/news/browser/news_listing.py]: Maybe use `{'portal_type': 'ftw.news.News'}` as default query?

**Comment by jone** [comment]
👍 looks good to me.
@mbaechtold please take look too since you understand it better than me 😉

File: ftw/news/browser/news_listing.py (hunk 2 above)
Comment (maethu) [inline ftw/news/browser/news_listing.py]: I decided to use the default query... of the inherited class --> `return super(NewsListingPortlet, self).get_query()
`

File: ftw/news/portlets/news_archive_portlet.py (hunk 1 above)
Comment (maethu) [inline ftw/news/portlets/news_archive_portlet.py]: Guess this is no necessary... there no special chars in the values.

**Comment by maethu** [comment]
@jone updated

**Pull request merged**